Bump dependencies to add GCP cloud logging exporter

damemi opened the pull request:
**Description:**
This imports an exporter for GCP logs

**Link to tracking Issue:** https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/8249

**Testing:** Manual and automated unit and integration tests in [GoogleCloudPlatform/opentelemetry-operations-go](https://github.com/GoogleCloudPlatform/opentelemetry-operations-go/tree/main/exporter/collector)

**Documentation:** Readme updated with new fields and sample configs

damemi committed:
Add GCP cloud logging exporter

codeboten reviewed (commented):
This is adding new functionality to the exporter, please add a note to the changelog

damemi commented:
@codeboten I updated the changelog, is there anything I need to do to get the tests to run again?

codeboten reviewed (commented):
> @codeboten I updated the changelog, is there anything I need to do to get the tests to run again?

The test run automatically on every commit so you should be ok there. Please resolve the conflicts and we'll get this merged 👍

damemi commented:
> > @codeboten I updated the changelog, is there anything I need to do to get the tests to run again?
> 
> The test run automatically on every commit so you should be ok there. Please resolve the conflicts and we'll get this merged 👍

Updated, thanks. btw it looks like first-time contributors need a maintainer to approve the workflows, at least that's the message I see. Just a heads up in case that's an unintentional setting

pmm-sumo commented on processor/schemaprocessor/go.mod:
```diff
@@ -5,7 +5,6 @@ go 1.17
 require (
 	github.com/stretchr/testify v1.7.1
 	go.opentelemetry.io/collector v0.50.1-0.20220429151328-041f39835df7
-	go.opentelemetry.io/collector/model v0.50.0
```
I think this and a couple of other changes in unrelated components are unintentional? I think it is better to remove them to not pollute the PR

damemi replied:
that's a good point, this change just came from running `make gotidy` from the root after I rebased to resolve conflicts in `exporter/googlecloudexporter/go.mod` and `processor/schemaprocessor/go.sum`. Not sure where this came from

damemi replied:
I get the same diff running gotidy from `main` as well. Is this not supposed to be run with each change?

damemi replied:
Updated to remove the unrelated changes from this PR

codeboten reviewed (commented):
Some tests are still failing:

```
make -C ./exporter/googlecloudexporter test
make[2]: Entering directory '/home/runner/work/opentelemetry-collector-contrib/opentelemetry-collector-contrib/exporter/googlecloudexporter'
go test -race -v -timeout 300s --tags="" ./...
=== RUN   TestLoadConfig
    config_test.go:60: 
        	Error Trace:	config_test.go:60
        	Error:      	Not equal: 
        	            	expected: &googlecloudexporter.Config{ExporterSettings:config.ExporterSettings{id:config.ComponentID{typeVal:"googlecloud", nameVal:"customname"}}, Config:collector.Config{ProjectID:"my-project", UserAgent:"opentelemetry-collector-contrib {{version}}", TraceConfig:collector.TraceConfig{ClientConfig:collector.ClientConfig{Endpoint:"test-trace-endpoint", UseInsecure:true, GetClientOptions:(func() []option.ClientOption)(nil)}, AttributeMappings:[]collector.AttributeMapping(nil)}, MetricConfig:collector.MetricConfig{ClientConfig:collector.ClientConfig{Endpoint:"test-metric-endpoint", UseInsecure:true, GetClientOptions:(func() []option.ClientOption)(nil)}, Prefix:"prefix", SkipCreateMetricDescriptor:true, KnownDomains:[]string{"googleapis.com", "kubernetes.io", "istio.io", "knative.dev"}, InstrumentationLibraryLabels:true, CreateServiceTimeSeries:false, CreateMetricDescriptorBufferSize:10, ServiceResourceLabels:true, ResourceFilters:[]collector.ResourceFilter(nil), CumulativeNormalization:true, EnableSumOfSquaredDeviation:false}, LogConfig:collector.LogConfig{ClientConfig:collector.ClientConfig{Endpoint:"", UseInsecure:false, GetClientOptions:(func() []option.ClientOption)(nil)}, DefaultLogName:""}}, TimeoutSettings:exporterhelper.TimeoutSettings{Timeout:20000000000}, QueueSettings:exporterhelper.QueueSettings{Enabled:true, NumConsumers:2, QueueSize:10}, RetrySettings:exporterhelper.RetrySettings{Enabled:true, InitialInterval:10000000000, MaxInterval:60000000000, MaxElapsedTime:600000000000}}
        	            	actual  : &googlecloudexporter.Config{ExporterSettings:config.ExporterSettings{id:config.ComponentID{typeVal:"googlecloud", nameVal:"customname"}}, Config:collector.Config{ProjectID:"my-project", UserAgent:"opentelemetry-collector-contrib {{version}}", TraceConfig:collector.TraceConfig{ClientConfig:collector.ClientConfig{Endpoint:"test-trace-endpoint", UseInsecure:true, GetClientOptions:(func() []option.ClientOption)(nil)}, AttributeMappings:[]collector.AttributeMapping(nil)}, MetricConfig:collector.MetricConfig{ClientConfig:collector.ClientConfig{Endpoint:"test-metric-endpoint", UseInsecure:true, GetClientOptions:(func() []option.ClientOption)(nil)}, Prefix:"prefix", SkipCreateMetricDescriptor:true, KnownDomains:[]string{"googleapis.com", "kubernetes.io", "istio.io", "knative.dev"}, InstrumentationLibraryLabels:true, CreateServiceTimeSeries:false, CreateMetricDescriptorBufferSize:10, ServiceResourceLabels:true, ResourceFilters:[]collector.ResourceFilter(nil), CumulativeNormalization:false, EnableSumOfSquaredDeviation:false}, LogConfig:collector.LogConfig{ClientConfig:collector.ClientConfig{Endpoint:"", UseInsecure:false, GetClientOptions:(func() []option.ClientOption)(nil)}, DefaultLogName:""}}, TimeoutSettings:exporterhelper.TimeoutSettings{Timeout:20000000000}, QueueSettings:exporterhelper.QueueSettings{Enabled:true, NumConsumers:2, QueueSize:10}, RetrySettings:exporterhelper.RetrySettings{Enabled:true, InitialInterval:10000000000, MaxInterval:60000000000, MaxElapsedTime:600000000000}}
        	            	
        	            	Diff:
        	            	--- Expected
        	            	+++ Actual
        	            	@@ -37,3 +37,3 @@
        	            	    ResourceFilters: ([]collector.ResourceFilter) <nil>,
        	            	-   CumulativeNormalization: (bool) true,
        	            	+   CumulativeNormalization: (bool) false,
        	            	    EnableSumOfSquaredDeviation: (bool) false
        	Test:       	TestLoadConfig
--- FAIL: TestLoadConfig (0.01s)
=== RUN   TestCreateDefaultConfig
    factory_test.go:31: 
        	Error Trace:	factory_test.go:31
        	Error:      	Received unexpected error:
        	            	type "Config" from package "github.com/open-telemetry/opentelemetry-collector-contrib/exporter/googlecloudexporter" has invalid config settings: type "Config" from package "github.com/GoogleCloudPlatform/opentelemetry-operations-go/exporter/collector" has invalid config settings: type "MetricConfig" from package "github.com/GoogleCloudPlatform/opentelemetry-operations-go/exporter/collector" has invalid config settings: mapstructure tag not present on field "EnableSumOfSquaredDeviation"
        	Test:       	TestCreateDefaultConfig
```

damemi commented:
Updated to pull in test fix from https://github.com/GoogleCloudPlatform/opentelemetry-operations-go/pull/380

damemi commented on exporter/googlecloudexporter/testdata/config.yaml:
```diff
@@ -24,6 +24,7 @@ exporters:
       skip_create_descriptor: true
       endpoint: test-metric-endpoint
       use_insecure: true
+      cumulative_normalization: false
```
@dashpole this look right? It got the test to pass. I'm assuming we want to test non-default here

dashpole replied:
Yep.  LGTM

Commit:
Merge branch 'main' into logs-collector-exporter
